src/realm/sync/client.cpp

```diff
@@ -1626,15 +1508,34 @@ void SessionWrapper::check_progress()
     REALM_ASSERT(!m_finalized);
     REALM_ASSERT(m_sess);
 
-    if (!m_progress_handler && m_upload_completion_handlers.empty() && m_sync_completion_handlers.empty())
+    // Check if there's anything which even wants progress or completion information
+    bool has_progress_handler = m_progress_handler && m_reliable_download_progress;
```

Comment:
Is there a valid reason outside of the unit tests to not have a progress handler at this point?

Reply:
No, m_progress_handler is always set when using SyncSession and it's only null in the sync unit tests.